static/js/services/db-sync.js

```diff
@@ -56,6 +56,12 @@ var _ = require('underscore'),
           })
           .then(function() {
             return keys;
+          })
+          .catch(function(err) {
+            // allow some replication otherwise the user can get stuck
+            // unable to fix their own configuration
+            $log.error('Error fetching sync options - using with minimum options', err);
+            return [ ALL_KEY ];
```

Comment:
I don't think that would work. On line 38 we set `var keys = [ ALL_KEY ];` and then later on we push values on to `keys` which would effectively change the value of the "constant".